File: fc-chat/demo/part3/ch3/03-2-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/EnterRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;

public class EnterRequest extends BaseRequest {

  private final ChannelId channelId;

  @JsonCreator
  public EnterRequest(@JsonProperty("channelId") ChannelId channelId) {
    super(MessageType.ENTER_REQUEST);
    this.channelId = channelId;
  }

  public ChannelId getChannelId() {
    return channelId;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/EnterRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId

class EnterRequest @JsonCreator constructor(@JsonProperty("channelId") channelId: ChannelId) :
    BaseRequest(MessageType.ENTER_REQUEST) {
    private val channelId: ChannelId = channelId

    fun getChannelId(): ChannelId {
        return channelId
    }
}
